### ISSUE TITLE
choosing a location crashes app

### ISSUE THREAD

**Issue by votacom** [votacom/octalsuntime#32]
There has been a report of the app crashing whenever choosing a location from the locations list, effectively having the clock always show the Greenwich time.

Status: waiting for logs.

**Comment by theAkito** [comment]
I added a single location. When adding it and selecting it again, I did not encounter any crash.

**Comment by votacom** [comment]
> I added a single location. When adding it and selecting it again, I did not encounter any crash.

It seems to me like you are referring to a different issue than this one. Your issue was #17 as has already been fixed.

This issue is about merely selecting a location from the list already crashing the app. Unfortunetaly, I am not able to reproduce it on the hardware and emulators I have, so I would rely on a full error log from an affected user.

**Comment by theAkito** [comment]
> It seems to me like you are referring to a different issue than this one. Your issue was #17 as has already been fixed.

> I did not encounter any crash.

I don't have an issue. Just wanted to provide feedback, that this issue is not happening on every phone. 😄

**Comment by PhiB9** [comment]
Hi,
this error also occures on my phone. Hope this helps:

```
type: crash
osVersion: google/raven/raven:13/TP1A.221105.002/2022112500:user/release-keys
package: at.manuelbichler.octalsuntime:5
process: at.manuelbichler.octalsuntime
processUptime: 2283 + 183 ms

java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1271436516, result=-1, data=Intent { dat=geo: }} to activity {at.manuelbichler.octalsuntime/at.manuelbichler.octalsuntime.ClockActivity}: java.lang.IllegalArgumentException: The given URI geo:0%2C0%3Fq%3D51%2C523251%2C6%2C925270(Bottrop) does not match the geo:0,0?q=lat,lng(label) format.
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5353)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5392)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2309)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7904)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
Caused by: java.lang.IllegalArgumentException: The given URI geo:0%2C0%3Fq%3D51%2C523251%2C6%2C925270(Bottrop) does not match the geo:0,0?q=lat,lng(label) format.
	at at.manuelbichler.octalsuntime.model.Location$Companion.fromUri(Location.kt:24)
	at at.manuelbichler.octalsuntime.LocationsActivity$SelectLocationContract.parseResult(LocationsActivity.kt:145)
	at at.manuelbichler.octalsuntime.LocationsActivity$SelectLocationContract.parseResult(LocationsActivity.kt:137)
	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:418)
	at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:375)
	at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:793)
	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:164)
	at android.app.Activity.dispatchActivityResult(Activity.java:8660)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5346)
	... 15 more

```

**Comment by votacom** [comment]
seems like the geo: URL generated by your systems uses comma (,) as a decimal delimiter, while a decimal point is expected in parsing. This likely because the String.format function uses the system's default locale, which in your cases is possibly German.

I could reproduce the issue by setting Android's system language to German.

**Comment by votacom** [comment]
This bug has been fixed in release v0.2.1. This updated app version should be available in F-Droid in a couple of days.
